hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -67,11 +108,15 @@ enum Term extends Statement:
   case Handle(lhs: LocalSymbol, rhs: Term, args: List[Term],
     derivedClsSym: ClassSymbol, defs: Ls[HandlerTermDefinition], body: Term)
   
-  lazy val symbol: Opt[Symbol] = this match
+  def symbol: Opt[Symbol] = this match
+    case ref: Ref if ref.resSym.nonEmpty => ref.resSym
     case Ref(sym) => S(sym)
-    case sel: SynthSel => sel.sym
     case sel: Sel => sel.sym
+    case sel: SynthSel => sel.sym
     case sel: SelProj => sel.sym
+    case app: App => app.sym
+    case tyApp: TyApp if tyApp.sym.nonEmpty => tyApp.sym
+    case TyApp(prefix, _) => prefix.symbol
```

Comment:
TyApp is similar to Ref. It always has the symbol from its LHS, except in the case that there is an implicit application on the TyApp. E.g., 

```fs
fun foo[A](using ...): module M
foo[Int]
```

The symbol of the Ref of foo is `foo` but the symbol of the TyApp of Ref of foo is `M`.

Reply:
I really don't like that the official symbol _changes_ over the phases of the compiler. Fix it like for references?